src/main/java/com/amazon/ion/impl/Ion11Writer.kt

```diff
@@ -0,0 +1,153 @@
+
+import com.amazon.ion.Decimal
+import com.amazon.ion.IonType
+import com.amazon.ion.Timestamp
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.time.Instant
+import java.time.LocalDate
+
+/**
+ * Writes Ion 1.1 data to an output source.
+ *
+ * This interface allows the user to write Ion data without being concerned about which output format is being used.
+ */
+interface Ion11Writer {
+
+    /**
+     * Indicates that writing is completed and all buffered data should be written and flushed as if this were the end
+     * of the Ion data stream. For example, an Ion binary writer will finalize any local symbol table, write all
+     * top-level values, and then flush.
+     *
+     * This method may only be called when all top-level values are completely written and (`stepped out`)[stepOut].
+     *
+     * Implementations should allow the application to continue writing further top-level values following the semantics
+     * for concatenating Ion data streams.
+     */
+    fun finish()
+
+    /**
+     * Closes this stream and releases any system resources associated with it.
+     * If the stream is already closed then invoking this method has no effect.
+     *
+     * If the cursor is between top-level values, this method will [finish] before closing the underlying output stream.
+     * If not, the resulting data may be incomplete and invalid Ion.
+     */
+    fun close()
+
+    /** Returns true if the writer is currently in a struct (indicating that field names are required). */
+    fun isInStruct(): Boolean
+
+    /** Returns the current depth of containers the writer is at.  This is 0 if the writer is at top-level. */
+    fun depth(): Int
+
+    /**
+     * Writes the Ion 1.1 IVM. IVMs can only be written at the top level of an Ion stream.
+     * @throws com.amazon.ion.IonException if in any container.
+     */
+    fun writeIVM()
+
+    /** Writes one annotation for the next value. */
+    fun writeAnnotations(annotation0: Long)
+
+    /** Writes two annotations for the next value. */
+    fun writeAnnotations(annotation0: Long, annotation1: Long)
+
+    /** Writes three or more annotations for the next value. */
+    fun writeAnnotations(annotation0: Long, annotation1: Long, vararg annotations: Long)
+
+    /** Writes one annotation for the next value. */
+    fun writeAnnotations(annotation0: String)
+
+    /** Writes two annotations for the next value. */
+    fun writeAnnotations(annotation0: String, annotation1: String)
+
+    /** Writes three or more annotations for the next value. */
+    fun writeAnnotations(annotation0: String, annotation1: String, vararg annotations: String)
+
+    /**
+     * Writes the field name for the next value. Must be called while in a struct and must be called before [writeAnnotations].
+     * @throws com.amazon.ion.IonException if annotations are already written for the value or if not in a struct.
+     */
+    fun writeFieldName(text: String)
```

Comment:
When will this be written inline vs as a symbol ID?

Reply:
Implementation detail. The raw writer will write the string method as inline field name and the long method as a SID. A configuration-driven, managed writer may choose to intern the field name in the symbol table, but it also depends on what sort of struct the writer is currently in.